src/python/pants/backend/rust/goals/tailor.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.backend.rust.target_types import RustPackageTarget
+from pants.core.goals.tailor import (
+    AllOwnedSources,
+    PutativeTarget,
+    PutativeTargets,
+    PutativeTargetsRequest,
+)
+from pants.engine.fs import PathGlobs, Paths
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.util.dirutil import group_by_dir
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class PutativeRustTargetsRequest(PutativeTargetsRequest):
+    pass
+
+
+@rule(level=LogLevel.DEBUG, desc="Determine candidate Rust targets to create")
+async def find_putative_rust_targets(
+    request: PutativeRustTargetsRequest, all_owned_sources: AllOwnedSources
+) -> PutativeTargets:
+    putative_targets = []
+
+    all_cargo_toml_files = await Get(Paths, PathGlobs, request.path_globs("Cargo.toml"))
+    unowned_cargo_toml_files = set(all_cargo_toml_files.files) - set(all_owned_sources)
```

Comment:
Oh, wouldn't all cargo toml files be unowned now?

Reply:
True